presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTask.java

```diff
@@ -880,7 +876,15 @@ private void failTask(Throwable cause)
             log.debug(cause, "Remote task %s failed with %s", taskStatus.getSelf(), cause);
         }
 
-        abort(failWith(getTaskStatus(), FAILED, ImmutableList.of(toFailure(cause))));
+        TaskStatus failedTaskStatus = failWith(getTaskStatus(), FAILED, ImmutableList.of(toFailure(cause)));
+        // Transition task to failed state without waiting for the final task info returned by the abort request.
+        // The abort request is very likely not to succeed, leaving the task and the stage in the limbo state for
```

Comment:
Curious why issue get raised by recoverable grouped execution? :)

Reply:
I see. Because failed task doesn't fail the whole query. Now if we don't update the task info , the failed task might stuck in RUNNING state :)

Reply:
Before the recoverable grouped execution the behaviour was "all or nothing". As long as any task fails - the entire query fails. And if the query is failed - the client doesn't wait for the final task info: https://github.com/prestodb/presto/blob/8d5d5e67e1e2276e9e2a1fc02f471e6d0a020c89/presto-main/src/main/java/com/facebook/presto/server/protocol/Query.java#L474.

In case of recoverable execution the task failure doesn't necessarily mean query failure. If a task fails the lifespan is rescheduled to a different task. 

However, even if all lifespans are done, the query cannot transition to FINISHED before every task receives the final task info. Since the node is (most likely) down the task will never get the final task info before timeout (10 minutes) (https://github.com/prestodb/presto/blob/5e3dec392d146925c4de9be0bbe47bed45020a22/presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTask.java#L864) effectively preventing the query from transitioning to FINISHED state. 

As a solution @shixuan-fan added a shortcut that used to transition a failed task right away to failed state, setting the final task info to "failed". Generally it makes perfect sense, as there's no reason to wait for 10 minutes for something that is know to be failed. 

However, the same method is used for `abort`. Abort is called when the query is finished, but some tasks are still running. Sometimes it is a legit behaviour (e.g.: SCAN -> LIMIT where LIMIT can terminate the query prematurely), and sometimes it is just racy (e.g.: query finishes before the coordinator receives the final task info for a task from a worker). Although it might be less important to receive correct task statistics for the first case (since it is inherently racy), for the second case it is pretty important. Otherwise stats could be very inconsistent.

Reply:
> Still , the task should eventually fail once all the abort retries finished right -- rather than stuck in RUNNING forever?

Yes. But current timeout is 10 minutes. Having your query stuck for 10 minutes is generally not desirable.

Reply:
Does this also address the issue where sometimes there are no stats available for the failure stage of a query?

Reply:
@rschlussel I'm not sure. It is likely to be unrelated though.